inc/rack.class.php

```diff
@@ -441,7 +441,7 @@ static function showForRoom(DCRoom $room) {
 
       $blueprint = "";
       $blueprint_ctrl = "";
-      if (strlen($room->fields['blueprint'])) {
+      if (strlen(($room->fields['blueprint'] ?? ""))) {
```

Comment:
Checking emptyness using `strlen` is not the best way to do IMHO.
```suggestion
      if (!empty($room->fields['blueprint'])) {
```